Add documentation for core module

matjazv opened the pull request:
### What was the problem?

This PR resolves LISK-2527.

### How was it solved?

Added comprehensive documentation for the SDK's `core` module, including detailed guides for `Client`, `Wallet`, `Transaction`, `Chains`, `Onramp` and `Utils` modules.

Each module now has step-by-step setup instructions, code examples, common use cases and cross-references to related modules.

Also added a main README with quick start guide and module overview.

### How was it tested?

sameersubudhi commented on packages/panna-sdk/src/core/docs/utils.md:
```diff
@@ -0,0 +1,342 @@
+
+  // Amount validation
+  const numAmount = parseFloat(amount);
+  if (!amount || numAmount <= 0) {
+    errors.push('Amount must be greater than 0');
+  }
```
I think, we should be able to post with 0 value as well but not negative.

matjazv replied:
Done.

sameersubudhi commented on packages/panna-sdk/src/core/docs/utils.md:
```diff
@@ -0,0 +1,342 @@
+- Explore the [Wallet Module](./wallet.md) for user authentication and account management
+- Check the [Transaction Module](./transaction.md) for sending tokens and smart contract interactions
+- Review the [Client Module](./client.md) for SDK initialization and configuration
+- Learn about [Chains Module](./chains.md) for network configuration
```
Include onramp here?

sameersubudhi commented on packages/panna-sdk/src/core/docs/wallet.md:
````diff
@@ -0,0 +1,278 @@
+Get authentication working in 2 minutes:
+
+```ts
+import {
+  createPannaClient,
+  createAccount,
+  prepareLogin,
+  login
+} from 'panna-sdk';
+
+// 1. Initialize SDK
+const client = createPannaClient({ clientId: 'your-client-id' });
+const ecosystem = { id: 'ecosystem.lisk', partnerId: 'your-partner-id' };
````
```suggestion
import {
  EcosystemId,
  createPannaClient,
  createAccount,
  prepareLogin,
  login
} from 'panna-sdk';

// 1. Initialize SDK
const client = createPannaClient({ clientId: 'your-client-id' });
const ecosystem = { id: EcosystemId.LISK, partnerId: 'your-partner-id' };
```

sameersubudhi commented on packages/panna-sdk/src/core/docs/wallet.md:
````diff
@@ -0,0 +1,278 @@
+### Basic Social Login
+
+```ts
+import { socialLogin } from 'panna-sdk';
+
+const ecosystem = { id: 'ecosystem.lisk', partnerId: 'your-partner-id' };
````
```suggestion
import { EcosystemId, socialLogin } from 'panna-sdk';

const ecosystem = { id: EcosystemId.LISK, partnerId: 'your-partner-id' };
```

sameersubudhi commented on packages/panna-sdk/src/core/docs/wallet.md:
```diff
@@ -0,0 +1,278 @@
+- [Transaction Module](./transaction.md) - Send transactions
+- [Chain Configuration](./chains.md) - Multi-chain support
+- [Fiat Onramp](./onramp.md) - Crypto purchases
+- [Utils Module](./utils.md) - Helpful utilities
```
Include client here?

matjazv replied:
Done.

matjazv commented on packages/panna-sdk/src/core/README.md:
````diff
@@ -0,0 +1,175 @@
+## Quick Start
+
+```typescript
+import { createPannaClient, createAccount, login, lisk } from 'panna-sdk';
````
I like Sameer's idea to export modules only and then call function from modules like  `client.createPannaClient({ clientId })`.

matjazv replied:
Yes, I agree to have new issue for this because there'll be quite few changes.

matjazv commented on packages/panna-sdk/src/core/docs/chains.md:
```diff
@@ -0,0 +1,242 @@
+
+### Lisk Mainnet
+
+Production network for Lisk applications.
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/onramp.md:
```diff
@@ -0,0 +1,207 @@
+      // 2. Prepare purchase
+      const session = await onRampPrepare({
+        client: this.client,
+        chainId: 1135,
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/onramp.md:
```diff
@@ -0,0 +1,207 @@
+    const id = sessionId || localStorage.getItem('onramp_session');
+    if (!id) throw new Error('No session ID found');
+
+    return await onRampStatus({ id, client: this.client });
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/transaction.md:
```diff
@@ -0,0 +1,223 @@
+const client = createPannaClient({ clientId: 'your-client-id' });
+const account = createAccount({ partnerId: 'your-partner-id' });
+
+// Send 1 LSK
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/transaction.md:
```diff
@@ -0,0 +1,223 @@
+  chain: lisk,
+  address: '0x...', // Token contract address
+  method: 'function transfer(address to, uint256 amount)',
+  params: [recipientAddress, toWei('100')]
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/transaction.md:
```diff
@@ -0,0 +1,223 @@
+- Explore [Chains Module](./chains.md) for configuring networks and RPC endpoints
+- Review [Utils Module](./utils.md) for balance queries and token conversions
+- Learn about [Wallet Module](./wallet.md) for user account management and authentication
+- Check [Onramp Module](./onramp.md) for integrating fiat-to-crypto purchases
```
Done.

matjazv commented on packages/panna-sdk/src/core/README.md:
```diff
@@ -0,0 +1,175 @@
+  client,
+  chain: lisk,
+  to: merchantAddress,
+  value: toWei('10') // 10 LSK
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/onramp.md:
```diff
@@ -0,0 +1,207 @@
+# Onramp Module
+
+The Onramp module enables fiat-to-crypto conversions through multiple payment providers. The module supports country-specific provider selection and comprehensive purchase flow management.
```
Done.

matjazv commented on packages/panna-sdk/src/core/docs/onramp.md:
```diff
@@ -0,0 +1,207 @@
+    }
+  }
+
+  return providers[0] || 'stripe';
```
Done!

matjazv commented on packages/panna-sdk/src/core/docs/client.md:
```diff
@@ -0,0 +1,220 @@
+import { createPannaClient, type PannaClient } from 'panna-sdk';
+
+const client = createPannaClient({
+  clientId: process.env.NEXT_PUBLIC_PANNA_CLIENT_ID!
```
See my reply above.

matjazv commented on packages/panna-sdk/src/core/docs/chains.md:
```diff
@@ -0,0 +1,242 @@
+  id: 12345,
+  rpc: 'https://my-custom-rpc.com',
+  nativeCurrency: {
+    name: 'Custom Token',
+    symbol: 'CTK',
+    decimals: 18
```
Inside this interface we're providing a native token (used for gas fees) for that chain and not tokens which can be deployed later like LSK token.
That's why we don't need to provide it's address because it's built into the protocol itself.

But to not confuse developers/users, I changed it to ETH.

matjazv commented on packages/panna-sdk/src/core/docs/onramp.md:
```diff
@@ -0,0 +1,207 @@
+const session = await onRampPrepare({
+  client,
+  chainId: 1135, // Lisk mainnet
+  tokenAddress: '0x0000000000000000000000000000000000000000', // Native ETH
```
I've created a test script and function `getTokenFiatPrices` returns `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` for ETH address.

I've updated `onramp.md` to use `NATIVE_TOKEN_ADDRESS`.

And also fixed `onramp.ts` file and examples inside because ThirdWeb functions always return `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` for ETH address. I've created short script and tested function `onRampPrepare`.

matjazv commented on packages/panna-sdk/src/core/docs/utils.md:
```diff
@@ -0,0 +1,342 @@
+    return groups;
+  },
+  {} as Record<string, typeof nfts>
+);
```
I've fixed the example.

Also created a test script, minted NFT for test address and this is output:
```
NFT Grouping Test Script
Testing fixes for PR #68, Comment #33

Testing NFT Grouping Functionality...

======================================================================
✓ Client created successfully

Test Configuration:
  Chain: Lisk Sepolia (4202)
  Address: 0x742d35Cc6634C0532925a3b844Bc9e7595f0bEb2

======================================================================
Step 1: Fetching NFT collection data...
======================================================================
✓ NFT data fetched successfully

Result structure:
  Total collectibles: 1
  Metadata - Count: 1
  Metadata - Has next page: false

======================================================================
Step 2: Testing NFT grouping logic (FIXED CODE)...
======================================================================
✓ NFT grouping completed successfully

Collections found: 1

======================================================================
Step 3: Displaying collection summary...
======================================================================

Collection: Lazy NFT (LN)
  Address: undefined
  Type: erc-1155
  Unique tokens: 1
  Total instances owned: 1
  Sample instances:
    1. ID: 1231233
       Name: N/A
       Image Type: unknown

======================================================================
✓ TEST PASSED - NFT grouping working correctly!
======================================================================
```

matjazv commented:
> * Should we move the readme files to the corresponding directory and rename to `README.md` instead of placing them in `packages/panna-sdk/src/core/docs/`? This would present a nice README on the GH UI when users navigate through the codebase

This is a good idea!
I've moved documents accordingly and updated the links.

matjazv commented on packages/panna-sdk/src/core/transaction/README.md:
```diff
@@ -0,0 +1,224 @@
+  chain: lisk,
+  to: recipientAddress,
+  value: toWei('1'),
+  gasPrice: BigInt(20000000000) // 20 gwei
```
Done!

matjazv commented on README.md:
```diff
@@ -26,6 +26,27 @@ This is a TypeScript monorepo for developing a Panna SDK.
+- **[Client](./packages/panna-sdk/src/core/client/README.md)** - SDK initialization and configuration
+- **[Wallet](./packages/panna-sdk/src/core/wallet/README.md)** - User authentication and account management
+- **[Transaction](./packages/panna-sdk/src/core/transaction/README.md)** - Blockchain transaction handling
+- **[Chains](./packages/panna-sdk/src/core/chains/README.md)** - Network configuration
```
Done.
